src/tagmanager/tm_parser.c

```diff
@@ -818,6 +818,28 @@ void tm_parser_verify_type_mappings(void)
 }
 
 
+/* determine anonymous tags from tag names only when corresponding
+ * ctags information is not available */
+gboolean tm_parser_is_anon_name(TMParserType lang, gchar *name)
+{
+	guint i;
+	char dummy;
+
+	if (lang == TM_PARSER_C || lang == TM_PARSER_CPP)
+	{
+		return sscanf(name, "anon_%*[a-z]_%u%c", &i, &dummy) == 1 ||
```

Comment:
Does sscanf really understand regexes? This is also new over the old code in `tm_tag_is_anon()`

Reply:
Yes on [Linux](https://man7.org/linux/man-pages/man3/scanf.3.html) but IIUC its not portable, especially `a-z` meaning lower case ASCII alphabetic, see [here](https://en.cppreference.com/w/c/io/fscanf)

Reply:
This code actually was there before the new cxx parser was merged. In the cxx pull request I changed it to `__anon` detection only but then thought there might be some old tag files around so better to use the previous checks too (plus now we get anonymous tags in this format again). Since it worked before, I guess it's OK.

Reply:
Agree